### PR TITLE
ORC-27: Add support for proleptic Gregorian calendar

### PR DESCRIPTION
**Pull request by omalley**
For better support of dates before 1600AD. This is a work in progress, I still need to fix the predicate push down. 

It would also be good to have the C++ writer set the proleptic calendar flag.

### CODE REVIEW

**Comment by omalley** [comment]
The CI will be broken until we release storage api 2.7.1.

**Comment by omalley** [comment]
The predicate push down and file and stripe statistics work now. I think this patch is pretty close.

**Comment by jcamachor** [comment]
LGTM, +1

File: proto/orc_proto.proto [hunk 1]
```diff
@@ -340,6 +340,15 @@ message Encryption {
   optional KeyProviderKind keyProvider = 4;
 }
 
+enum CalendarKind {
+  UNKNOWN_CALENDAR = 0;
+   // The Java default calendar changes from Julian to Gregorian
+   // in 1583.
```
Comment (abstractdog) [inline proto/orc_proto.proto]: just noticed, actually it's 1582, we can fix this minor comment issue later

**Pull request closed without merging**